tests/llmobs/test_llmobs_eval_metric_writer.py

```diff
@@ -69,6 +77,18 @@ def test_send_metric_bad_api_key(mock_writer_logs):
     )
 
 
+@pytest.mark.vcr_logs
+def test_send_metric_no_api_key(mock_writer_logs):
```

Comment:
I think this has to do with the above fix, great catch but let's move this to a separate fix PR

Reply:
ref this comment: https://github.com/DataDog/dd-trace-py/pull/11535#discussion_r1899545469

It's not necessarily a fix, but just changing the behavior of where we log in `submit_evaluation_for` vs `submit_evaluation`